server/app/controllers/ti/TrustedIntermediaryController.java

```diff
@@ -124,7 +121,41 @@ public Result dashboard(
   }
 
   @Secure(authorizers = Authorizers.Labels.TI)
-  public Result editClient(Long accountId, Http.Request request) {
+  public Result showAddClientForm(Long id, Http.Request request) {
+    Optional<CiviFormProfile> civiformProfile = profileUtils.currentUserProfile(request);
+    if (civiformProfile.isEmpty()) {
+      return unauthorized();
+    }
+    Optional<TrustedIntermediaryGroupModel> trustedIntermediaryGroup =
+        accountRepository.getTrustedIntermediaryGroup(civiformProfile.get());
+    if (trustedIntermediaryGroup.isEmpty()) {
+      return notFound();
+    }
+    if (!trustedIntermediaryGroup.get().id.equals(id)) {
+      return unauthorized();
+    }
```

Comment:
Question (non-blocking): Do we do similar authorization checks in other methods in this controller? Is there an easy way to pull them out into a helper function that can be reused?